herd/AArch64Sem.ml

```diff
@@ -2625,57 +2622,85 @@ module Make
             !!!!(read_reg_ord rA ii >>= fun addr ->
             (store_m addr rs ii >>|
             post_kr rA addr kr ii))
-
-        | I_LDR_SIMD(var,r1,rA,kr,s) ->
+        | I_LDR_SIMD(var,r1,rA,MemExt.Reg(v,kr,sext,s)) ->
+            let access_size = tr_simd_variant var in
+            get_ea_reg rA v kr sext s ii >>= fun addr ->
+            simd_ldr access_size addr r1 ii >>= B.next1T
+        | I_LDR_SIMD(var,r1,rA,MemExt.Imm (k,Idx)) ->
+            let access_size = tr_simd_variant var in
+            get_ea_idx rA k ii >>= fun addr ->
+            simd_ldr access_size addr r1 ii >>= B.next1T
+        | I_LDR_SIMD(var,r1,rA,MemExt.Imm (k,PreIdx)) ->
             let access_size = tr_simd_variant var in
-            get_ea rA kr s ii >>= fun addr ->
+            get_ea_preindexed rA k ii >>= fun addr ->
             simd_ldr access_size addr r1 ii >>= B.next1T
-        | I_LDR_P_SIMD(var,r1,rA,k) ->
+        | I_LDR_SIMD(var,r1,rA,MemExt.Imm (k,PostIdx)) ->
             let access_size = tr_simd_variant var in
             read_reg_ord rA ii >>= fun addr ->
             simd_ldr access_size addr r1 ii >>|
             post_kr rA addr (K k) ii >>= B.next2T
         | I_LDUR_SIMD(var,r1,rA,k) ->
-            let access_size = tr_simd_variant var and
-            k = K (match k with Some k -> k | None -> 0) in
-            (get_ea rA k S_NOEXT ii >>= fun addr ->
-            simd_ldr access_size addr r1 ii) >>= B.next1T
+            let access_size = tr_simd_variant var in
+            get_ea rA (K k) S_NOEXT ii >>= fun addr ->
+            simd_ldr access_size addr r1 ii >>= B.next1T
         | I_LDAPUR_SIMD(var,r1,rA,k) ->
-            let access_size = tr_simd_variant var and
-            k = K (match k with Some k -> k | None -> 0) in
-            (get_ea rA k S_NOEXT ii >>= fun addr ->
-            simd_ldar access_size addr r1 ii) >>= B.next1T
-        | I_STR_SIMD(var,r1,rA,kr,s) ->
             let access_size = tr_simd_variant var in
-            simd_str access_size rA r1 kr s ii
-        | I_STR_P_SIMD(var,r1,rA,k) ->
+            get_ea rA (K k) S_NOEXT ii >>= fun addr ->
+            simd_ldar access_size addr r1 ii >>= B.next1T
+        | I_STR_SIMD(var,r1,rA,MemExt.Reg (v,kr,sext,s)) ->
+            let access_size = tr_simd_variant var in
+            get_ea_reg rA v kr sext s ii >>= fun addr ->
+            simd_str access_size addr r1 ii
```

Comment:
This has the unfortunate consequence of changing the intra-instruction relations. So I would suggest instead doing this:
```suggestion
            let ma = get_ea_reg rA v kr sext s ii in
            simd_str access_size ma r1 ii
```

Reply:
And the same applies to all store variants in this PR. The loads are ok.